### ISSUE TITLE
spineData param is required, 

### ISSUE THREAD

**Issue by whizzkid** [pixijs/spine#467]
I'm trying to get pixi 6 and spine runtime 4.1 working, 
I would love to just load pixi and spine using <script> tags, but I'd also like to have the typedefinitions so that my editor (Rider, or vsCode) can do code completion/insight.

Apparantly the whole world is using webpack so I'm trying that now,  (but I REALLY rather not have 80mb+ of crap in every project folder)

ALMOST had it working, but the spine loader doesn't work. and vsCode and rider now give different errors on the same script.... spent a whole day on it and am now more confused than when I started...
I'm giving up....

Is there anyone who can provide me with a complete WORKING typescript example project that just does the following and has working code completion/code insight in vscode or Rider?
- preferably without webpack
- maybe with the project code dumped into 1 output file with tsConfig's  outFile param.

I don't need a build in webserver, or auto-reload, or anything,   just a working project... that my editor understands.

if it run's this, I can take it from there..
```
const app = new PIXI.Application();
document.body.appendChild(app.view);

app.loader
    .add('spineCharacter', 'src/assets/spine/Monster2.json')
    .load(function (loader, resources) {
        const animation = new Spine(resources.spineCharacter.spineData);
        app.stage.addChild(animation);        
        app.start();
    });
```

**Comment by ivanpopelyshev** [comment]
I know your pain, that's why there's special section in README:
What you need is UMD build : https://github.com/pixijs/spine#vanilla-js-umd-build

basically, old way is namespaces. Everything goes into `PIXI.pluginName.Class` 
new way is imports. `import {Class} from 'pluginName';`

for old way you need usual `<script src` and UMD builds.

**Comment by ivanpopelyshev** [comment]
Unfortunately, people who know only how to do JS through webpack dont actually, and old guys like I and you dont know that its possible to just install module and copy single UMD build from `node_modules/...` whatever, maybe even put this command in a good old gruntfile

That's how we have those kind of threads :)

**Comment by ivanpopelyshev** [comment]
Oh, right, code completion. Sorry, cant guarantee that because TS declarations are made with imports in mind ) If you find the way, or the way to make global typescript definitions using api-extractor, please share it. 

I personally use Idea/WebStorm/Rider and i didnt see problems related to that, I dont know why it doesnt work for you

**Comment by bencresty** [comment]
> Apparantly the whole world is using webpack so I'm trying that now, (but I REALLY rather not have 80mb+ of crap in every project folder)

That 'crap' in your project folder and Webpack makes sure you don't have all kinds of unused crap in your production output.

Eventhough there are no browsers working with Typescript, never was and never will be, you still want to use Typescript, but don't want to setup a chain to translate Typescript to Javascript... and want some kind of magic thing that magically works without using the software needed to do so....

My advise would be to start being realistic; 
either learn to setup tools to do what you're after, invest time if what you're stating is actually true (for example, when using PNPM you DON't have all this 'crap' in a project folder, but you're using references) or stop complaining, throw overboard your wishes and rethink if you got what it takes to be a developer.

**Comment by whizzkid** [comment]
Right.. I'll log that away.. and after rethinking my carreer choices and coming to the conclusion that the past 26 years of being a fulltime programmer was kinda fun, I think I'm gonna continue to be one.

anyways.

@ivanpopelyshev  

Ok so I'll give the UMD build a try later when I get this working properly via webpack.
I got the spine to load, but something is weird.

It seems that the spineParser doesn't get called when the asset is loaded. (using the example from pixi-spine's repo page)
it also doesnt work when you manually do     ```SpineParser.registerLoaderPlugin();```

But when I do ```app.loader.use(SpineParser.use);``` then the parser is actually added to the Loader._afterMiddleware list and it actually parses the Spinedata and I can display Spineboy on screen.  yay..

But that doesn't seem to be normal does  it?

Here's my lil project:
[pixi_spine_webpack.zip](https://github.com/pixijs/spine/files/9801185/pixi_spine_webpack.zip)

to run:  
```
npm i
npm run start
```

I tried using the spineboy example https://pixijs.io/examples/#/plugin-spine/spineboy-pro.js and found more weirdness.
Clicking on the spineboy with this setup does nothing...   
If I add a Graphic circle and click that,  it does work. 

If I make spineBoyPro interactive. I get loads of errors.

![image](https://user-images.githubusercontent.com/323372/196193596-dd0acd11-dcba-4eff-ad87-649c8cb1d88e.png)

So something seems off :)

**Comment by ivanpopelyshev** [comment]
At that moment, pixi-spine requires  pixi 6.1.0 minimum due to specific things related to typings in webpack-oriented modules )))))

You installed pixi 6.0.4 6.1.0, those deps are fighting each other.

**Comment by whizzkid** [comment]
Updated pixi to 6.1.0, deleted package.lock, node modules, rebuild,  had to update  webpack-cli to 4.10.0 because it suddenly gave errors on options.foreach..    

But I still need to do  app.loader.use(SpineParser.use);  otherwise the spine won't load,
and if i make spineboy interactive, the same errors appear..

Can you take a look at the project I uploaded in my previous post?
I must be doing something wrong, but I dont know what. :|

**Comment by jomin398** [comment]
```
loader.add('spineCharacter', rawSkelUrl, { metadata: { spineAtlas: spineAtlas } });
    loader.load(function (that, resources) {
        //thrown same error on here
        const animation = new PIXI.spine.Spine(resources.spineCharacter.spineData);
        console.log(animation)
    })
```

same error to me ...  how to add `loader.use(SpineParser.use);` in `umd.js`?
